music_assistant/helpers/smart_fades.py

```diff
@@ -0,0 +1,879 @@
+"""Smart Fades - Object-oriented implementation with intelligent fades and adaptive filtering."""
+
+# TODO: Figure out if we can achieve shared buffer with StreamController on full
+# current and next track for more EQ options.
+# TODO: Refactor the Analyzer into a metadata controller after we have split the controllers
+# TODO: Refactor the Mixer into a stream controller after we have split the controllers
```

Comment:
Let's make sure we come up with a plan soon. This is a core feature so we either bake it into the core controller (as submodule) or we simply invent something like providertype.core_module so we can isolate everything in provider-like compartments

Reply:
Let's discuss soon 👌